File: src/main/java/com/example/demo/school/Student.java
package com.example.demo.school;

public class Student {
    String id;
    String name;
    int english;
    int math;
    static int pass = 60;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public void print(){
        int average = getAverage();
        System.out.println(name + "\t" + english + "\t" + math + "\t"
                        + getAverage() + "\t" + ((average >= pass) ? "PASS" : "FAILED"));
        char grading = 'F';
        switch (average / 10) {
            case 10 :
            case 9 :
                grading = 'A';
                break;
            case 8 :
                grading = 'B';
                break;
            case 7 :
                grading = 'C';
                break;
            case 6 :
                grading = 'D';
                break;
            default:
                grading = 'F';
        }
        System.out.println("\t"+ grading);

        /*if (getAverage() >= 60) {
            System.out.println("\tPASS");
        }else{
            System.out.println("\tFAILED");
        }*/
    }

    public int highest(){
//        if (english > math ){
//            max = english;
//        }else{
//            max = math;
//        }
        return (english > math ) ? english : math ;
    }

    public int getAverage(){
        return (english + math) / 2;
    }
}

File: src/main/kotlin/com/example/kotlin/Student.kt
package com.example.kotlin

import java.util.*

fun main() {
    val stu = Student("Annie", 80, 66)
    stu.print()
    val t = 123
    println("test: $t")
    println("Highest score : ${stu.highest()}")



//    userInput()
}



class Student(var name : String?, var english : Int, var math : Int) {
    fun print() {
/*
        print(
            name + "\t" + english + "\t" + math + "\t"
                    + getAverage() + "\t"
                    + if (getAverage() >= 60) "PASS" else "FAILED")
*/
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")

    }

    fun getAverage() = (english + math) / 2

    fun passOrFailed() = if (getAverage() >= 60) "PASS" else "FAILED"

    fun grading() =  when (getAverage()) {
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }


/*
    fun grading(): Char {
        var grading = when (getAverage()) {
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
        }
        return grading
    }
*/


    fun highest() = if (english > math) {
            println("english")
            english
        } else {
            println("math")
            math
        }
/*        var max = if (english > math) {
            println("english")
            english
        } else {
            println("math")
            math
        }
        return max
*/

/*      if(english > math) {
        max = english
         }else{
        max =math
*/


    fun check() = println(name?.length)

/*
    fun check() {
        println(name?.length)
    }
*/

    private fun userInput() {
        var scanner = Scanner(System.`in`)
//    print("please enter your name: ");
//    var name = scanner.next();
        var name = null
        print("enter your english score: ")
        var english = scanner.nextInt();
        print("enter your math score: ")
        var math = scanner.nextInt();
        val stu = Student(name, english, math)
        stu.print();
        stu.check();
    }
}

